### PR TITLE
fix(no_std_executor): Ensure thread-safe initialization of Executor in current function

### PR DESCRIPTION
**Pull request by younes-io**
fix(no_std_executor): Ensure thread-safe initialization of Executor in current function

- Introduced atomic operations to prevent concurrent initialization of Executor
- Utilized compare_exchange_weak for safe and single initialization
- Enhanced overall thread safety and memory safety in a no_std environment

## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] There are no Merge commits in this Pull Request. Ockam repo maintains a linear commit history. We merge Pull Requests by rebasing them onto the develop branch. Rebasing to the latest develop branch and force pushing to your Pull Request branch is okay.
- [x] I have read and accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have read and accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/.github/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam/blob/develop/.github/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam](https://github.com/build-trust/ockam) repository. The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam/edit/develop/.github/CONTRIBUTORS.csv) file in the github web UI and create a separate Pull Request, this will mark the commit as verified.

### CODE REVIEW

File: implementations/rust/ockam/ockam_executor/src/executor.rs [hunk 1]
```diff
@@ -12,21 +12,31 @@ use ockam_core::compat::sync::Arc;
 use ockam_core::compat::task::Wake;
 
 /// Returns current executor.
-/// WARNING: TODO this is not thread-safe
+static INIT: AtomicBool = AtomicBool::new(false);
+static mut EXECUTOR: UnsafeCell<MaybeUninit<Executor>> = UnsafeCell::new(MaybeUninit::uninit());
+
 pub fn current() -> &'static Executor<'static> {
-    static INIT: AtomicBool = AtomicBool::new(false);
```
Comment (younes-io) [inline implementations/rust/ockam/ockam_executor/src/executor.rs:17]: BTW, whoever is going to review this; I got this code out of that function to make it a singleton. Is that OK ?

**Comment by younes-io** [comment]
Can someone help me fix the commit msg ?

**Review by SanjoDeundiak — CHANGES_REQUESTED** [review]
Could you please take a look at why CI fails?

**Comment by younes-io** [comment]
Closing because this keeps failing and I get a million emails a day... and I don't have access to the CI/CD

**Pull request closed without merging**